packages/vitest/src/types/mocker.ts

```diff
@@ -1,6 +1,8 @@
+type Promisable<T> = T | Promise<T>
+
 export type MockFactoryWithHelper<M = unknown> = (
-  importOriginal: <T extends M>() => Promise<T>
-) => any
+  importOriginal: <T extends M = M>() => Promise<T>
+) => Promisable<M | null>
```

Comment:
1. `null` is never allowed
2. Let's return `Partial<M>` - it's allowed to return only a part of the module. I think it's better to throw an error with a nice stack trace like we already do if you forgot to return an export

Reply:
Thank you for the review. It's all reflected.